src/components/NWCEditor.tsx

```diff
@@ -0,0 +1,522 @@
+import {
+    createForm,
+    getValue,
+    required,
+    setValue,
+    SubmitHandler
+} from "@modular-forms/solid";
+import { BudgetPeriod, Contact, NwcProfile } from "@mutinywallet/mutiny-wasm";
+import {
+    createMemo,
+    createResource,
+    For,
+    Match,
+    ResourceFetcher,
+    Show,
+    Switch
+} from "solid-js";
+
+import {
+    AmountEditable,
+    AmountSats,
+    Avatar,
+    Button,
+    Checkbox,
+    InfoBox,
+    KeyValue,
+    LoadingShimmer,
+    TextField,
+    TinyText,
+    VStack
+} from "~/components";
+import { useI18n } from "~/i18n/context";
+import { useMegaStore } from "~/state/megaStore";
+import { fetchNostrProfile } from "~/utils";
+
+type BudgetInterval = "Day" | "Week" | "Month" | "Year";
+export type BudgetForm = {
+    connection_name: string;
+    auto_approve: boolean;
+    budget_amount: string; // modular forms doesn't like bigint
+    interval: BudgetInterval;
+    profileIndex?: number;
+    nwaString?: string;
+};
+
+type FormMode = "createnwa" | "createnwc" | "editnwc";
+type BudgetMode = "fixed" | "editable";
+
+function parseNWA(nwaString?: string) {
+    if (!nwaString) return undefined;
+    const nwa = decodeURI(nwaString);
+    if (nwa) {
+        // Examples:
+        // nostr+walletauth://a957bc527d4b7cea5134308412719fa675671ed38eb313adcf89b96c6982480e?relay=wss%3A%2F%2Frelay.damus.io%2F&secret=a2522b4c6d6ae729&required_commands=pay_invoice&budget=21%2Fday&identity=04c915daefee38317fa734444acee390a8269fe5810b2241e5e6dd343dfbecc9
+        // nostr+walletauth://c95053d4df36104c7d1a007c4c46fd195920d268527c21d5474b1842ef2e4170?relay=wss%3A%2F%2Frelay.damus.io%2F&secret=39133ed58614e67b&required_commands=pay_invoice&budget=21%2Fday&identity=0d6c8388dcb049b8dd4fc8d3d8c3bb93de3da90ba828e4f09c8ad0f346488a33
```

Comment:
these 2 examples are exactly the same besides different keys. not a huge deal but don't see the point of having both